src/main/java/TaskList.java

```diff
@@ -0,0 +1,159 @@
+import java.io.BufferedReader;
+import java.util.ArrayList;
+import java.util.stream.Stream;
+
+public class TaskList {
+    private ArrayList<Task> tasksArr = new ArrayList<Task>();
+    BufferedReader bufferedReader;
+    TaskList(BufferedReader bufferedReader){
+        this.bufferedReader = bufferedReader;
+        populateTaskList();
+    }
+
+    public TaskList() {
+
+    }
+
+    /**
+     * Use the Buffered File provided in the constructor to read the file and populate the list with the stored
+     * tasks
+     */
+    private void populateTaskList() {
+        Stream<String> linesStream = bufferedReader.lines();
+        linesStream.forEach(line -> {
+            String[] lineSplit = line.split("//");
+            String taskType = lineSplit[0];
+            Task currTask = new Task("temp");
+            if (taskType.equals("T")){ // Todo
+                boolean isDone = lineSplit[1].equals("1");
+                String taskName = lineSplit[2];
+                currTask = new Todo(taskName);
+                if (isDone){
+                    currTask.done();
+                }
+            } else if (taskType.equals("D")) { // Deadline
+                boolean isDone = lineSplit[1].equals("1");
+                String taskName = lineSplit[2];
+                String endDate = lineSplit[3];
+                currTask = new Deadline(taskName, endDate);
+                if (isDone){
+                    currTask.done();
+                }
+            } else if (taskType.equals("E")) {
+                boolean isDone = lineSplit[1].equals("1");
+                String taskName = lineSplit[2];
+                String eventDate = lineSplit[3];
+                currTask = new Event(taskName, eventDate);
+                if (isDone){
+                    currTask.done();
+                }
+            } else {
+                System.out.println("Invalid syntax in duke.txt for " + taskType);
+            }
+            tasksArr.add(currTask);
+        });
+    }
+
+    /**
+     * Get the string representation of the full list to write to the file and store the data
+     * @return string
+     */
+    public String taskListString () {
```

Comment:
Will it be better to do all the formatting in Todo/Deadline/Event classes (eg. having a storageFormat() method in these classes which return a string representation of the format in which you want the task to be stored) to achieve better abstraction?